src/main/java/gov/nasa/pds/api/engineering/controllers/MyBundlesApiController.java

```diff
@@ -231,18 +177,17 @@ private Products getCollectionChildren(String lidvid, int start, int limit, List
 		 else return new ResponseEntity<Products>(HttpStatus.NOT_IMPLEMENTED);
 	}
 
-    @SuppressWarnings("unchecked")
 	private Products getProductChildren(String lidvid, int start, int limit, List<String> fields, List<String> sort, boolean onlySummary) throws IOException
     {
-    	
     	if (!lidvid.contains("::")) lidvid = this.productBO.getLatestLidVidFromLid(lidvid);
-    	
     	MyBundlesApiController.log.info("request bundle lidvid, children of products: " + lidvid);
-    	GetRequest getBundleRequest = new ElasticSearchRegistrySearchRequestBuilder().getGetProductRequest(lidvid);
+
+    	int iteration=0,wsize=0;
     	HashSet<String> uniqueProperties = new HashSet<String>();
-    	List<String> productLidvids = new ArrayList<String>();
+    	List<String> clidvids = this.getRefLidCollection(lidvid, limit);
```

Comment:
it is unclear to me here as well: we are asking for product children of bundle from start to start+limit, I don't see how that can match with the collection pagination, so we can not use the limit parameter here

Reply:
As state above, it is used a random number more than anything else. It could be fixed to 10 or 3 or 50 or 100. Much bigger than 100 and elasticsearch returns an error regularly.

It is an interesting question of what to set the limit of the collection of a bundle to when looking for the product. 

The idea is that if the user wants to wait for bigger chunks like 100 products then also getting 100 collections is better than 10 collections then having to ask again for the other 90 in the case of one product per collection.